File: backup/Invest-Invest/Invest-Invest-Repository/src/main/java/com/illunex/invest/invest/entity/Invest.java
package com.illunex.invest.invest.entity;

import com.illunex.invest.invest.entity.enumable.InvestState;

import javax.persistence.*;
import java.util.List;

@Entity
public class Invest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idx;
    Long userIdx;
    Long companyIdx;
    InvestState investState;

    @OneToMany
    List<Judge> judges;             // 심사위원
    @OneToMany
    List<ExamItem> examItems;       // 심사항목
}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/Invest.kt
package com.illunex.invest.invest.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "invest")
data class Invest (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null,
    var userIdx: Long? = null,
    var companyIdx: Long? = null,

    @OneToMany(mappedBy = "evaluate", cascade = [CascadeType.ALL])
    var judges: List<Judge>
)


